tests/api2/test_api_key_crud.py

```diff
@@ -0,0 +1,49 @@
+import contextlib
+
+from middlewared.test.integration.utils import call, client
+
+
+@contextlib.contextmanager
+def api_key(allowlist):
+    key = call("api_key.create", {"name": "Test API Key", "allowlist": allowlist})
+    try:
+        yield key
+    finally:
+        call("api_key.delete", key["id"])
+
+
+def test_has_key_after_creation_but_not_read():
+    api_key = call("api_key.create", {"name": "Test", "allowlist": []})
```

Comment:
It's a little confusing to use the name of a function as the name of a local variable imo